CorsixTH/Src/th_map.cpp

```diff
@@ -485,7 +485,7 @@ bool level_map::load_from_th_file(const uint8_t* pData, size_t iDataLength,
 
       if (!(pData[5] & 1)) {
         pNode->flags.passable = true;
-        if (!(pData[7] & 16)) {
+        if (!(pData[7] & 16) && pNode->iParcelId != 0) {
```

Comment:
Albert also states in https://github.com/CorsixTH/CorsixTH/issues/1991#issuecomment-1221563728 it ideally needs an afterload to handle this change. What that would look like I'm not sure... There's probably a method to request a refresh of map data from lua.

Reply:
Was thinking of `Map:setCellFlags` but generally this requires each tile to be given unless there's a way to efficiently pull all pond tiles from a map and then apply.

Reply:
Iterating over them and selection pond tiles is the best you can do here. Even in CPP that's what you need to do.